tests/dialects/test_gpu.py

```diff
@@ -111,6 +111,31 @@ def test_block_id():
     assert block_id.dimension is dim
 
 
+def test_dealloc():
+
+    typ = memref.MemRefType.from_element_type_and_shape(
+        builtin.Float32Type(), [10, 10, 10])
+    alloc = AllocOp.get(typ, is_async=True)
+
+    # For pyright
+    assert alloc.asyncToken is not None
+
+    dealloc = DeallocOp.get(buffer=alloc.result,
+                            async_dependencies=[alloc.asyncToken],
+                            is_async=True)
+
+    assert dealloc.asyncToken is not None
+    assert isinstance(dealloc.asyncToken.typ, AsyncTokenType)
+    assert dealloc.buffer is alloc.result
+    assert dealloc.asyncDependencies == tuple([alloc.asyncToken])
+
+    sync_dealloc = DeallocOp.get(buffer=alloc.result)
```

Comment:
Feels like bad practice to dealloc the same thing two times here.

Reply:
It is, I'm just testing the constructor. I can try to add a second alloc.

Reply:
Done!